callout/src/main/java/com/google/apigee/ProtoMessageBuilders/MessageContextProtoMessageBuilder.java

```diff
@@ -0,0 +1,160 @@
+ * MessageContext} object.
+ */
+public class MessageContextProtoMessageBuilder {
+  /**
+   * Builds a MessageContext Protocol Buffer Message using a {@link MessageContext} object.
+   *
+   * @param messageContext {@link MessageContext} object used to construct the MessageContext
+   *     Protocol Buffer Message.
+   * @return MessageContext Protocol Buffer Message
+   */
+  public static Execute.MessageContext buildMessageContextProto(MessageContext messageContext) {
+    Execute.MessageContext.Builder messageContextBuilder = Execute.MessageContext.newBuilder();
+    if (messageContext.getMessage(FlowContext.TARGET_REQUEST) != null) {
+      messageContextBuilder.setTargetRequestMessage(
+          buildMessageProto(messageContext.getMessage(FlowContext.TARGET_REQUEST)));
+    }
+    if (messageContext.getMessage(FlowContext.PROXY_REQUEST) != null) {
+      messageContextBuilder.setProxyRequestMessage(
+          buildMessageProto(messageContext.getMessage(FlowContext.PROXY_REQUEST)));
+    }
+    if (messageContext.getMessage(FlowContext.TARGET_RESPONSE) != null) {
+      messageContextBuilder.setTargetResponseMessage(
+          buildMessageProto(messageContext.getMessage(FlowContext.TARGET_RESPONSE)));
+    }
+    if (messageContext.getMessage(FlowContext.PROXY_RESPONSE) != null) {
+      messageContextBuilder.setProxyResponseMessage(
+          buildMessageProto(messageContext.getMessage(FlowContext.PROXY_RESPONSE)));
+    }
+    if (messageContext.getErrorMessage() != null) {
+      messageContextBuilder.setErrorMessage(buildMessageProto(messageContext.getErrorMessage()));
+    }
+    return messageContextBuilder.build();
+  }
+
+  /**
+   * Builds a Message Protocol Buffer Message using a {@link Message} object.
+   *
+   * @param message {@link Message} object used to construct the Message Protocol Buffer Message.
+   * @return Message Protocol Buffer Message
+   */
+  private static Execute.Message buildMessageProto(Message message) {
+    Execute.Message.Builder messageBuilder = Execute.Message.newBuilder();
+    if (message.getContent() != null) {
+      messageBuilder.setContent(ByteString.copyFrom(message.getContent(), StandardCharsets.UTF_8));
+    }
+    if (message.getHeaderNames() != null) {
+      messageBuilder.putAllHeaderMap(buildHeaderMap(message));
+    }
+    if (message.getQueryParamNames() != null) {
+      messageBuilder.putAllQueryParamMap(buildQueryParametersMap(message));
+    }
+    messageBuilder.putAllFlowVariables(new HashMap<>());
+    return messageBuilder.build();
+  }
+
+  /**
+   * Builds the FlowMapValue for the Flow Variable map in the Message Protocol Buffer Message.
+   *
+   * @param value Either a String or FlowInfo object to store in the Flow Variable map.
+   * @return FlowMapValue to be stored in the Flow Variable map in the Message Protocol Buffer
+   *     Message.
+   */
+  private static Execute.Message.FlowMapValue buildFlowMapValue(Object value) {
```

Comment:
christine called this out earlier as well, it's not used right now since we can't extract the map directly yet, but i just left that in there for use in the future once we can do that.

Reply:
comment on this and remove private so it can be tested?